src/mots/cli.py

```diff
@@ -0,0 +1,240 @@
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+"""
+usage: mots [-h] [--debug] {init,module,validate,clean,query} ...
+
+main command line interface for mots
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --debug               enable debug output
+
+commands:
+  {init,module,validate,clean,query}
+    init                initialize mots configuration in repo
+    module              module_operations
+    validate            validate mots config for current repo
+    clean               clean mots config for current repo
+    query               query the module directory
+"""
+import argparse
+from datetime import datetime
+import logging
+import sys
+from pathlib import Path
+
+from mots import module
+from mots.config import FileConfig
+from mots.directory import Directory
+from mots.logging import init_logging
+from mots.config import DEFAULT_CONFIG_FILEPATH
+
+from mots.utils import get_list_input
+
+logger = logging.getLogger(__name__)
+
+
+class CLI:
+    """Wrap methods and call with the correct arguments."""
+
+    @staticmethod
+    def init(args):
+        """Call `file_config.init` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.init()
+
+    @staticmethod
+    def ls(args):
+        """Call `module.ls` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        return module.ls(file_config.config["modules"])
+
+    @staticmethod
+    def show(args):
+        """Call `module.show` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        return module.show(file_config.config["modules"], args.module)
+
+    @staticmethod
+    def validate(args):
+        """Call `module.validate` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        return module.validate(file_config.config, args.repo_path)
+
+    @staticmethod
+    def clean(args):
+        """Call `module.clean` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        return module.clean(file_config)
+
+    @staticmethod
+    def query(args):
+        """Call `directory.query` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        directory = Directory(file_config)
+        directory.load()
+        result, rejected = directory.query(*args.paths)
+        for path in result:
+            modules = result[path]
+            module_names = ",".join([m.machine_name for m in modules])
+            owners = ",".join([",".join(m.owners) for m in modules])
+            sys.stdout.write(f"{path}:{module_names}:{owners}\n")
+
+    @staticmethod
+    def add(args):
+        """Prompt user to add a new module."""
+        params = {
+            "machine_name": input("Enter machine name of new module: "),
```

Comment:
This prompt may be hard to understand for people not familiar with the nomenclature: "what's the machine name? _My_ machine's name?" 😉